test/clients.go

```diff
@@ -38,8 +39,8 @@ type KubeClient struct {
 }
 
 // NewSpoofingClient returns a spoofing client to make requests
-func NewSpoofingClient(client *KubeClient, logf logging.FormatLogger, domain string, resolvable bool) (*spoof.SpoofingClient, error) {
-	return spoof.New(client.Kube, logf, domain, resolvable, Flags.IngressEndpoint)
+func NewSpoofingClient(client *KubeClient, transport *http.Transport, logf logging.FormatLogger, domain string, resolvable bool) (*spoof.SpoofingClient, error) {
```

Comment:
Why not accept roundtripper?

Reply:
Because DialContext needs to be modified.

Reply:
yeah but you can cast inside `accept interface, return struct` :-)

Reply:
Not sure this applies here. What do I do if it is not actually a http.Transport?
I really need an http.Transport, not an http.RoundTripper.

Reply:
Hmm, it simplifies a few things, but ohwell.
I just hate the cast in the `wait` function below.

Reply:
Me too, http.DefaultTransport should be an *http.Transport, not an http.RoundTripper.
`accept interface, return struct`, so much for returning structs :-|

Reply:
:-) https://en.wikipedia.org/wiki/Brad_Fitzpatrick -- sits in our building and maintains that library.